### DISCUSSION TITLE
Calculating the tortuosity from 3D images.

### DISCUSSION THREAD

**Discussion by li-ruidong** [PMEAL/porespy#1003] [Q&A]
Hi, I am new to porespy and want to figure out how to calculate the tortuosity. I followed the instructions in https://porespy.org/examples/simulations/tutorials/using_tortuosity_gdd.html but got nothing. Notably, I didn't have the properties of 'pore.left' and 'pore_right'. Very confused about how to use this excellent package as many tutorials are introduced based on artificial data. Can someone help me with this problem? Many thanks.

**Comment by jgostick** [comment]
The `tortuosity_gdd` function is a work in progress and the page probably shouldn't be there yet.  I recommend using [TauFactor2](https://github.com/tldr-group/taufactor).
